### ISSUE TITLE
Appveyor PR builds failing

### ISSUE THREAD

**Issue by PromoFaux** [PromoFaux/Matterhook.NET.MatterhookClient#16]
I've clearly set something up wrong in the Appveyor config. Builds for tags work with no issue, and branch builds appear to work.

It appears to relate to the AssemblyInfo

```text
obj\Release\net452\Matterhook.NET.MatterhookClient.AssemblyInfo.cs(18,59): warning CS7035: The specified version string does not conform to the recommended format - major.minor.build.revision [C:\projects\matterhook-net-matterhookclient\Matterhook.NET.MatterhookClient\Matterhook.NET.MatterhookClient.csproj]
obj\Release\net452\Matterhook.NET.MatterhookClient.AssemblyInfo.cs(22,55): error CS7034: The specified version string does not conform to the required format - major[.minor[.build[.revision]]] [C:\projects\matterhook-net-matterhookclient\Matterhook.NET.MatterhookClient\Matterhook.NET.MatterhookClient.csproj]
```

**Comment by rgomez90** [comment]
So, I think I`ve solved this, but I'm not really if I know why/how 😄

I`ve made a new fork of the project and new project in (my) appveyor to make test.

After a while in Appveyor's project configuration site I saw this

![image](https://user-images.githubusercontent.com/12009909/47451215-7b2fe400-d7c7-11e8-8f3f-44f0b5e93a71.png)

Since I am reaaaally very new to CI, I don't really know much about that, but I guess when activating this option, (like image says) a random string is appended to the build version number. 
The file numbers MUST BE numbers and must be able to be stored (afaik) in an `uint` (look [here](https://stackoverflow.com/a/37941296/5688800)).

So my guess is this option creates an UNVALID version number and is setted to the `version` variable in `appveyor.yml`.

I disabled the option  and it builds on pr as well, so I think that's the issue.

If so, with my little knowledege about CI, I guess we have 2 options: 

* Disable it
* Remove this "random" string somewhere before building

But I am sure there are more "sofisticated" (😄) ways of doing this... 
 
In my "little investigation" I noticed a few "strange" things (I dont say bugs since I dont really know as said about CI).

* The artifact built with this configuration was a package ending in **1.0.0** (hmmm?) it was build Nr. 6, so it should be 1.0.6?

* It should not be 1.3.6 instead of 1.0.6?

* When taking a look into the created DLL with the [Nuget Package Explorer](https://github.com/NuGetPackageExplorer/NuGetPackageExplorer) I noticed:

  * **Wrong version as well** 
![image](https://user-images.githubusercontent.com/12009909/47451999-681e1380-d7c9-11e8-81e7-df580812c17c.png)

  * **Wrong version numbers in created dlls**
![image](https://user-images.githubusercontent.com/12009909/47452086-a1ef1a00-d7c9-11e8-9dac-817101026765.png).

**Comment by PromoFaux** [comment]
Honestly I don't think I ever did the versioning correctly on this project. In the old days I'd just bump the version in `assemblyinfo.cs`, I'm a bit new to the `dotnet core` way of doing things!

**Comment by PromoFaux** [comment]
Fixed in #15
